apps/user-portal/src/components/overview/widgets/account-status-widget.tsx

```diff
@@ -29,28 +33,236 @@ import { ThemeIcon } from "../../shared";
+     *
+     * @param field - Field to check.
+     * @return {ProfileCompletionStatus}
+     */
+    const getFiledCompletionStatus = (field: any): ProfileCompletionStatus => {
+
+        const percentage = (field.completedCount / field.totalCount) * 100;
+
+        if (percentage <= UIConstants.ERROR_ACCOUNT_STATUS_UPPER_LIMIT) {
+            return ProfileCompletionStatus.ERROR;
+        } else if (percentage <= UIConstants.WARNING_ACCOUNT_STATUS_UPPER_LIMIT) {
+            return ProfileCompletionStatus.WARNING;
+        }
+
+        return ProfileCompletionStatus.SUCCESS;
+    };
+
+    /**
+     * Resolved the type of account status shield based on the completion status.
+     *
+     * @return {any}
+     */
+    const resolveStatusShield = () => {
+        const status = getProfileStatus();
+
+        if (status === ProfileCompletionStatus.SUCCESS) {
+            return AccountStatusShields.good;
+        } else if (status === ProfileCompletionStatus.ERROR) {
+            return AccountStatusShields.danger;
+        } else if (status === ProfileCompletionStatus.WARNING) {
+            return AccountStatusShields.warning;
+        }
+
+        return AccountStatusShields.good;
+    };
+
+    /**
+     * Generates the more info popup.
+     *
+     * @param attributes - Relevant attributes.
+     * @return {JSX.Element}
+     */
+    const generatePopup = (attributes: any): JSX.Element => (
+        ((attributes.completedAttributes
+            && attributes.completedAttributes.length
+            && attributes.completedAttributes.length > 0)
+            || (attributes.incompleteAttributes
+                && attributes.incompleteAttributes.length
+                && attributes.incompleteAttributes.length > 0))
+            ? (
+                <Popup
+                    trigger={ <Icon color="grey" name="info circle"/> }
+                    position="bottom center"
+                    className="list-content-popup"
+                    hoverable
+                    pinned
+                    content={ (
+                        <>
+                            {
+                                (attributes.completedAttributes
+                                    && attributes.completedAttributes.length
+                                    && attributes.completedAttributes.length > 0)
+                                    ? (
+                                        <>
+                                            <div className="header">
+                                                <Icon color="green" name="check circle"/>
+                                                { t("views:components.overview.widgets.accountStatus." +
+                                                    "completedFields") }
+                                            </div>
+                                            <ul>
+                                                {
+                                                    attributes.completedAttributes
```

Comment:
Can we get the display name of the attribute here instead of the attribute name?

Reply:
Display names have been added.